src/utils.ts

```diff
@@ -1254,3 +1255,28 @@ export async function request(
     req.end();
   });
 }
+
+/**
+ * A custom AbortController that aborts after a specified timeout.
+ *
+ * If `timeout` is undefined or \<=0, the abort controller never aborts.
+ *
+ * This class provides two benefits over the built-in AbortSignal.timeout() method.
+ * - This class provides a mechanism for cancelling the timeout
+ * - This class supports infinite timeouts by interpreting a timeout of 0 as infinite.  This is
+ *    consistent with existing timeout options in the Node driver (serverSelectionTimeoutMS, for example).
+ * @internal
+ */
+export class TimeoutController extends AbortController {
+  constructor(
+    timeout?: number,
+    private timeoutId = timeout && timeout > 0 ? setTimeout(() => this.abort(), timeout) : null
+  ) {
+    super();
+  }
+
+  clear() {
+    this.timeoutId && clearTimeout(this.timeoutId);
```

Comment:
I appreciate the brevity but mind switching to `if != null`? short circuiting in an expression is not as obvious as control flow